modules/git/blob_nogogit.go

```diff
@@ -102,7 +103,17 @@ func (b *blobReader) Read(p []byte) (n int, err error) {
 
 // Close implements io.Closer
 func (b *blobReader) Close() error {
+	if b.rd == nil {
+		return fs.ErrClosed
+	}
```

Comment:
Should it `panic` here instead? Calling `Close` multiple times is a coding error. I checked all calls to `DataAsync` and could not find another double close. A `panic` could prevent double closes in future while returning a (most of time) ignored error could hide it.

Reply:
No. None of Golang's other "Close" methods panics.

Reply:
Could it simply return "nil"? Most (all) errors for a Close method are meanless .... no caller cares about it.

Reply:
I think double Close should be supported. i.e. It's safe to invoke `file.Close` twice. So I think we should also allow invoking `Close` twice. It's convenient for some situations.

Reply:
Should we then add this rule to our coding guidelines?
> If a type implements `io.Closer`, calling `Close` multiple times must not fail and return a `nil` error.

Reply:
I guess it depends, for most cases: yes, `nil` is good enough.

But I guess there could be some special cases that the caller might be interested in the result.

Reply:
Then maybe relax the `nil` but specify that calling `Close` multiple times must not be undefined behaviour and must not panic?